students/FyfyNguyen/session04/mailroom2.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+"""
+Mailroom 2
+"""
+
+import sys
+import tempfile
+import pathlib
+from textwrap import dedent
+
+donor_db = {"Robyn Rihanna": [100, 200, 300],
+            "Ariana Grande": [2250, 4000, 1000],
+            "Beyonce Carter-Knowles": [150000, 3500, 25000],
+            "Aubrey Drake Graham": [15000, 5500.25, 1200],
+            "Justin Bieber": [2500, 250, 750.50]
+            }
+
+file_dir = pathlib.Path(tempfile.gettempdir())
+
+
+def print_donors():
+    """
+    Prints a list of exiting donors in donor_db
+    """
+    for donor in donor_db:
+        print(donor)
+
+
+def find_donor(name_entered):
+    """
+    Lookup donor in donor_db
+
+    :param: the name of the donor
+    :returns: the donor data structure -- None if donor not found in donor_db
+    """
+    donor_key = name_entered.strip().title()
+    return donor_db.get(donor_key)
+
+
+def add_donor(name_entered):
+    """
+    Add a new donor to donor_db
+
+    :param: the name of the donor
+    :returns: the new donor data structure
+    """
+    donor = (name_entered, [])
+    donor_db[name_entered.title()] = []
+    print("Successfully added new donor to database.")
+    add_donation(name_entered)
+
+
+def add_donation(name_entered):
+    """
+    Ask the user for donation amount and add it to donor_db
+
+    :param: the name of the donor
+    """
+    donation = float(input("Enter donation amount >>> "))
+    donor_db.setdefault(name_entered.title(), []).append(donation)
```

Comment:
This is a nice way to either create a new donro if there isn't one, or add to the one that's there if there is -- nice find!

However, in this case, I'd probably be more explicit about whether you are adding a new one or adding to an existing donor.